buku

```diff
@@ -529,19 +530,16 @@ class BukuDb:
             Path to database file.
         """
 
-        data_home = os.environ.get('XDG_DATA_HOME')
-        if not data_home:
-            if not os.environ.get('HOME'):
-                if sys.platform == 'win32':
-                    data_home = os.environ.get('APPDATA')
-                    if not data_home:
-                        return os.path.abspath('.')
-                else:
-                    return os.path.abspath('.')
-            else:
-                data_home = os.path.join(os.environ.get('HOME'), '.local', 'share')
-
-        return os.path.join(data_home, 'buku')
+        _get = os.environ.get
+        if _get('BUKU_DEFAULT_DBDIR'):
+            return os.path.abspath(_get('BUKU_DEFAULT_DBDIR'))
+        home_locations = [
+            _get('XDG_DATA_HOME'),
+            _get('HOME') and os.path.join(_get('HOME'), '.local', 'share'),
+            sys.platform == 'win32' and _get('APPDATA'),
+        ]
+        data_home = next((s for s in home_locations if s), None)
+        return (os.path.join(data_home, 'buku') if data_home else os.getcwd())
```

Comment:
This (after `BUKU_DEFAULT_DBDIR`) is the same logic as before but rewritten to be more straightforward